corgidrp/l2a_to_l2b.py

```diff
@@ -171,13 +171,33 @@ def correct_bad_pixels(input_dataset):
         not want to correct everything in the DQ extension
         - Different bad pixels in the DQ may be corrected differently.
 
-
     Args:
         input_dataset (corgidrp.data.Dataset): a dataset of Images (L2a-level)
+        bp_mask (corgidrp.data.BadPixelMap): Bad-pixel mask flagging all bad pixels
+            in that frame. Must be 0 (good) or 1 (bad) at every pixel.
 
     Returns:
-        corgidrp.data.Dataset: a version of the input dataset with bad pixels corrected
-    """
-
-    return None
-
+        corgidrp.data.Dataset: a version of the input dataset with bad detector
+        pixels and cosmic rays replaced by 0s
+ 
+    """
+
+    data = input_dataset.copy()
+    data_cube = data.all_data
+    dq_cube = data.all_dq
+
+    for i in range(data_cube.shape[0]):
+        # load CR mask
+        cr_mask = dq_cube[i]
+        # combine CR and BP masks
+        bp_cr_mask = (cr_mask+bp_mask[0].data).astype(int)
```

Comment:
Kevin brought up a good point in the thread for PR #93. Rather than just add things here, I think we want to break it into bits. I think in the long run it'll be safer and more robust.

Reply:
I think this can be done pretty easily with `numpy.packbits`, `numpy.unpackbits` and the `numpy.bitwise_or`.

Reply:
Sounds good. I understood that was a method that could be implemented later on. I'll implement it now

Reply:
@maxwellmb: Main has dq as 0 or 1: https://github.com/roman-corgi/corgidrp/blob/de9f4b00c6b8ffe4c2a26bfd2bb80b4cf090209f/corgidrp/data.py#L172

I'm not sure now which data type is the numpy array in dq by looking at data.py if it's not up to date with the DRP implementation document. Looking at other tests, dq are integers. 

If they are integers, are yiu suggesting to cast them into bits arrays, checking bit array values and performing operations with bits, to come back to integers when dq is stored again?

Reply:
> @maxwellmb: Main has dq as 0 or 1:
> 
> https://github.com/roman-corgi/corgidrp/blob/de9f4b00c6b8ffe4c2a26bfd2bb80b4cf090209f/corgidrp/data.py#L172

Yes, I think that line is out of date. Good catch! Do you mind updating it? 

> I'm not sure now which data type is the numpy array in dq by looking at data.py if it's not up to date with the DRP implementation document. Looking at other tests, dq are integers.

Yes, I _think_ the dq should be uint16

> 
> If they are integers, are you suggesting to cast them into bit arrays, checking bit array values and performing operations with bits, to come back to integers when dq is stored again?

Yup!

Reply:
https://numpy.org/doc/stable/reference/generated/numpy.unpackbits.html#numpy.unpackbits requires np.uint8 to work. It does not if it is np.uint16. I'll be using uint8.

Reply:
As for the doc string, we agreed to move it to a minor Issue: https://github.com/roman-corgi/corgidrp/issues/103

Reply:
Done. I resubmitted both l2a_to_l2b.py and test_badpixel.